lib/network.py

```diff
@@ -669,7 +671,8 @@ def maintain_sockets(self):
         # Responses to connection attempts?
         while not self.socket_queue.empty():
             server, socket = self.socket_queue.get()
-            self.connecting.remove(server)
+            if server in self.connecting:
+                self.connecting.remove(server)
             if socket:
```

Comment:
I didn't know that. So discard is like TryRemove in some other languages. Thanks.